src/test_rig_bluesky/plans.py

```diff
@@ -38,6 +51,44 @@ def spectroscopy(
     yield from bps.prepare(
         spectroscopy_detector, TriggerInfo(livetime=exposure_time), wait=True
     )
+    # TODO: This would be nicer if NDArrayBaseIO had a PortName signal
+    yield from bps.abs_set(spectroscopy_detector.fileio.nd_array_port, "D2.roistat")
+
+    rois = [
+        ROI(2, "Green", 880, 605, 150),
+        ROI(3, "Blue", 1665, 600, 150),
+        ROI(1, "Red", 95, 610, 150),
+    ]
+
+    params: list[NDAttributePv] = []
+    for roi in rois:
+        roistatn = spectroscopy_detector.roistat.channels[roi.channel]  # type: ignore
+        assert isinstance(roistatn, NDROIStatNIO)
+
+        yield from bps.abs_set(roistatn.name_, roi.name)
+        yield from bps.abs_set(roistatn.min_x, roi.start_x)
+        yield from bps.abs_set(roistatn.min_y, roi.start_y)
+        yield from bps.abs_set(roistatn.size_x, roi.size)
+        yield from bps.abs_set(roistatn.size_y, roi.size)
+        yield from bps.abs_set(roistatn.use, True)
```

Comment:
Must: These won't block until put complete so could lead to race conditions, you could instead:

```suggestion
        yield from bps.abs_set(roistatn.name_, roi.name, group="setup")
        yield from bps.abs_set(roistatn.min_x, roi.start_x, group="setup")
        yield from bps.abs_set(roistatn.min_y, roi.start_y, group="setup")
        yield from bps.abs_set(roistatn.size_x, roi.size, group="setup")
        yield from bps.abs_set(roistatn.size_y, roi.size, group="setup")
        yield from bps.abs_set(roistatn.use, True, group="setup")

        # This line blocks until all puts are complete
        yield from bps.wait(group="setup")
```

OR

```suggestion
        bps.mv(roistatn.name_, roi.name,
            bps.abs_set(roistatn.min_x, roi.start_x,
            bps.abs_set(roistatn.min_y, roi.start_y,
            bps.abs_set(roistatn.size_x, roi.size,
            bps.abs_set(roistatn.size_y, roi.size,
            bps.abs_set(roistatn.use, True,
            wait=True,
        )
```